src/main/kotlin/bakalarka/elearningplatform/controller/CourseController.kt

```diff
@@ -0,0 +1,74 @@
+package bakalarka.elearningplatform.controller
+
+import bakalarka.elearningplatform.model.Course
+import bakalarka.elearningplatform.model.TOPIC
+import bakalarka.elearningplatform.request.AddCourseChapterRequest
+import bakalarka.elearningplatform.request.AddCourseRequest
+import bakalarka.elearningplatform.service.ChapterService
+import bakalarka.elearningplatform.service.CourseService
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/courses")
+class CourseController(
+    var courseService: CourseService,
+    var chapterService: ChapterService
+) {
+
+    @GetMapping("/{courseId}")
+    fun getById(@PathVariable courseId: Long) = courseService.findById(courseId)
+
+    @GetMapping("/instructor")
+    @PreAuthorize("hasAuthority('update:courses')")
+    fun getByUserId() = courseService.findByUserId()
+
+    @GetMapping("")
+    fun findByTitle(@RequestParam(required = false) title: String?): MutableSet<Course> {
+        return if (title == null) {
+            courseService.getAll()
+        } else {
+            courseService.findByTitle(title)
+        }
+    }
+
+    @PutMapping("/{courseId}")
+    @PreAuthorize("hasAuthority('update:courses')")
+    fun updateCourse(
+        @PathVariable courseId: Long,
+        @RequestBody request: AddCourseRequest
+    ): ResponseEntity<Course> {
+        val response = courseService.updateCourse(courseId, request)
+        return if (response == null) {
+            ResponseEntity(HttpStatus.UNAUTHORIZED)
+        } else {
+            ResponseEntity.ok(response)
+        }
+    }
+
+    @GetMapping("/topics")
+    fun getTopics() = TOPIC.values()
```

Comment:
Toto sa niekde vyuziva? Ak hej, tak by som to presunul do separatneho controlleru - aj za cenu vytvorenia noveho controlleru :D